pubsub/topic.go

```diff
@@ -897,11 +909,21 @@ func (t *Topic) publishMessageBundle(ctx context.Context, bms []*bundledMessage)
 			opt.Resolve(&settings)
 		}
 		r := &publishRetryer{defaultRetryer: settings.Retry()}
-		res, err = t.c.pubc.Publish(ctx, &pb.PublishRequest{
-			Topic:    t.name,
-			Messages: pbMsgs,
-		}, gax.WithGRPCOptions(grpc.MaxCallSendMsgSize(maxSendRecvBytes)),
-			gax.WithRetry(func() gax.Retryer { return r }))
```

Comment:
I think readability here can be improved to actually conditionally build the gax.CallOptions depending of the `PublishSettings` and keep the call to `.Publish` appearing only once. Would be something like:

```go
gaxOpts := []gax.CallOption{
	gax.WithGRPCOptions(grpc.MaxCallSendMsgSize(maxSendRecvBytes)),
	gax.WithRetry(func() gax.Retryer { return r }),
}
if t.PublishSettings.shouldCompress(batchSize) {
	gaxOpts = append(gaxOpts, gax.WithGRPCOptions(grpc.UseCompressor(gzip.Name)))
}
res, err = t.c.pubc.Publish(ctx, &pb.PublishRequest{
	Topic:    t.name,
	Messages: pbMsgs,
}, gaxOpts...)
```		

Also separate the logic into the PublishSettings struct, so it can be easily tweaked later.
```go
func (ps *PublishSettings) shouldCompress(batchSize int) bool {
	return ps.EnableCompression && batchSize > ps.CompressionBytesThreshould
}
```